### DISCUSSION TITLE
Stuck in Generating WSA download link

### DISCUSSION THREAD

**Discussion by karlz97** [LSPosed/MagiskOnWSALocal#252] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/48748912/201510088-71af8225-ce48-4567-96ab-d53ccbcb73c0.png)
I have been waiting for 30 min, but nothing shows up. Is this normal? super confused. (also tried several times

**Comment by Howard20181** [comment, marked as answer]
It seems to be stuck requesting WSA download address from Microsoft server, I tested it locally with no problem. This could be a problem with Microsoft's servers or your network.

**Reply by karlz97** [reply]
Thanks for your kind and fast reply! The only thing I didn't follow the readme tutorial is I download the repo instead of clone it to local(which I think should be irrelavant to this?)

btw could you tell me why this project cannot provide built release? is that be cause the building is hardware-specified?

thx!

**Reply by karlz97** [reply]
I am using a quite usual US family network... werid... I will try using anther network tomorrow

thx

**Reply by Howard20181** [reply]
> Thanks for your kind and fast reply! The only thing I didn't follow the readme tutorial is I download the repo instead of clone it to local(which I think should be irrelavant to this?)
> 
> btw could you tell me why this project cannot provide built release? is that be cause the building is hardware-specified?
> 
> thx!

Having dealt with permission issues before, there was no problem downloading directly rather than cloning.

It is possible to build x86_64 and arm64 at any host, so it is not hardware specific.
Pre-built versions are not provided because secondary distribution of modified WSAs would be a copyright risk, and we don't know Microsoft's tolerance for modified WSAs. So this repo is launched as a tool.
And it is a question of who should distribute and where to store the pre-built version, so we do not provide it.

**Reply by karlz97** [reply]
Appreciate a lot for your reply and your work!
